File: silver/1735/c++/main.cpp
#include <iostream>

int main()
{
    int a, b, c, d, r;
    std::cin >> a >> b >> c >> d;

    int num0 = (a * d + c * b);
    int num1 = (b * d);

    r = 1;
    a = num0;
    b = num1;

    while (r > 0) {
        r = a % b;
        a = b;
        b = r;
    }

    std::cout << (num0 / a) << ' ' << (num1 / a);
    return 0;
}

File: silver/1735/python/main.py

if (__name__ == "__main__"):
    a, b = map(int, input().split())
    c, d = map(int, input().split())

    num0, num1 = (a*d + b*c), (b*d)

    r = 1
    a, b = (num0, num1)
    while (b):
        a, b = (b, a % b)
    print((num0 // a), num1 // a)




